Objective-C/CBLQueryResult.mm

```diff
@@ -55,7 +55,7 @@ - (instancetype) initWithResultSet: (CBLQueryResultSet*)rs
 #pragma mark - CBLArray
 
 - (NSUInteger) count {
-    CBL_LOCK(_rs.database) {
+    CBL_LOCK(_rs.query) {
```

Comment:
Instead of calling c4query_columnCount() here, I think it's better to move this to CBLQuery class itself so CBLQueryResult doesn't need to use c4Query at all.